src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -318,6 +342,7 @@ private string FormatEnumValueToString(string value, JavaScriptEncoder? encoder)
 
         internal override T ReadAsPropertyNameCore(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
         {
+            string? enumString = null;
```

Comment:
It is unfortunate that the main `Read` method calls into `ReadAsPropertyNameCore`, which is a method intended for dictionary key deserialization. Should `options.DictionaryKeyPolicy` be consulted here instead of the converter-level naming policy? Or does the converter level naming policy override `options.DictionaryKeyPolicy`?

Would it make sense to break up the method into two separate implementations? One for value deserialization and one for dictionary keys?

Reply:
The dictionary naming policy isn't applied on deserialization, just serialization. The initial changes were introducing a bug where the naming policy would be applied when deserializing dictionary keys. So yeah we do need to split the implementation and add more tests.